dg/workflow.md

```diff
@@ -0,0 +1,53 @@
+<frontmatter>
+  header: header.md
+  pageNav: 2
+  pageNavTitle: "Chapters of This Page"
+  siteNav: dg-nav.md
+  footer: footer.md
+</frontmatter>
+
+# Workflow of CATcher
+
+The following is the workflow that contributors need to follow when proposing changes to CATcher.
+
+1. [Submitting PRs](#submitting-prs)
+2. [Testing](#testing)
+    1. [Generating Spec Files](#generating-spec-files)
+
+## Submitting PRs
+
+When a pull request is merged, it should be accompanied by a helpful commit message. Hence, contributors are recommended to propose a commit message for their pull requests.
+[Here](https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message) are
+some good guidelines for crafting the commit message.
+
+## Testing
+
+We use [Scuri](https://github.com/gparlakov/scuri) to generate spec files for unit tests.
```

Comment:
I think we can remove the info on generating spec files with Scuri, as we haven't been using it much for our tests.